### PR TITLE
Also set new (non-beta/non-deprecated) labels in buildGenericLabels

### PR DESCRIPTION
**Pull request by codablock**
This fixes issues when an ASG has 0 nodes yet and node labels are built
using buildGenericLabels and the node-template labels.

Issues include (anti-)affinity and nodeSelectors for the given labels,
giving false-negative results for candidate nodes, which leads to ASGs
never scaling up.

### CODE REVIEW

**Comment by codablock** [comment]
I see that 1.21 is underway. Is there a chance to still get this into 1.21? It might also make sense to backport this to older versions, as it looks like all versions are affected by this.

**Comment by codablock** [comment]
After looking into other cloud providers, I realised that all cloud providers have this same issue. I decided to reimplement the solution in a generic way (based on updateDeprecatedTemplateLabels) and force-pushed this.

**Comment by gjtempleton** [comment]
Thanks for the PR and putting in the work to make it a generic solution.

There's a couple of AWS related conflicts after the merging of #3848, however other than it looks good to me from the AWS side.

**Comment by codablock** [comment]
@gjtempleton fyi, I've resolved the merge conflicts in the meantime.

**Comment by codablock** [comment]
@gjtempleton Hello. Just wanted to check if there is any update on planning to merge this one? My understand is that we're ready for merge.

**Comment by gjtempleton** [comment]
Sorry for the delay!

Thanks for the nudge. All looks good on the AWS side. 

You'll still need approval from one of the higher-level approvers of the wider CA codebase as well as mentioned by the CI-robot.
/assign aleksandra-malinowska
/lgtm

File: cluster-autoscaler/cloudprovider/gce/templates_test.go
Comment (elmiko) [inline cluster-autoscaler/cloudprovider/gce/templates_test.go:230]: minor nit, spacing format

**Review by elmiko — COMMENTED** [review]
overall this looks nice to me, just a minor spacing issue on the tests, otherwise i'm /lgtm

File: cluster-autoscaler/cloudprovider/gce/templates_test.go
Comment (codablock) [reply cluster-autoscaler/cloudprovider/gce/templates_test.go]: Fixed, squashed and rebased on master

**Comment by elmiko** [comment]
thanks @codablock !
/lgtm

**Review by feiskyer — APPROVED** [review]
LGTM

**Pull request merged**
